dwave/cloud/computation.py

```diff
@@ -748,23 +747,45 @@ def sampleset(self):
                                "Re-install the library with 'bqm' support.")
 
         # filter inactive variables from samples
-        variables = set(self.variables)
+        variables = self.variables
         samples = [[sample[v] for v in variables] for sample in self.samples]
 
         # infer vartype from problem type
         # note: KeyError on unknown problem types. BQMs should be handled above.
         vartype_from_problem_type = {'ising': 'SPIN', 'qubo': 'BINARY'}
         vartype = vartype_from_problem_type[self.problem_type]
 
-        # include timing in info
-        info = dict(timing=self.timing)
+        # include timing and id in info
+        info = dict(timing=self.timing, problem_id=self.id)
 
         sampleset = dimod.SampleSet.from_samples(
             (samples, variables), vartype=vartype,
             energy=self.energies, num_occurrences=self.occurrences,
             info=info, sort_labels=True)
 
-        self._result['sampleset'] = sampleset
+        # this means that samplesets retrieved BEFORE this function are called
+        # are not the same object as after, but it is a simpler implementation
+        self._result['sampleset'] = self._sampleset = sampleset
```

Comment:
The more complicated implementation would be something like
```
try:
    self._result['sampleset'] = self._sampleset
except AttributeError:
    self._result['sampleset'] = self._sampleset = sampleset
```
I didn't go with it because (I think) it's pretty confusing and because doing any reading in the main branch of the try-catch causes an infinite recursion error. But I could do it and make a long comment about it....

Reply:
No, I think it's fine as it is.